File: python/boj10866.py
'''
백준 10866. 덱

덱 구현하기
'''
from collections import deque
import sys
input = sys.stdin.readline
n = int(input())
queue = deque()
res = []
for _ in range(n):
    order = input().split()

    if order[0] == "push_back":
        queue.append(order[1])
    elif order[0] == "push_front":
        queue.appendleft(order[1])
    elif order[0] == "pop_front":
        if not queue:
            res.append(-1)
            continue
        res.append(queue.popleft())
    elif order[0] == "pop_back":
        if not queue:
            res.append(-1)
            continue
        res.append(queue.pop())
    elif order[0] == "size":
        res.append(len(queue))
    elif order[0] == "empty":
        if queue:
            res.append(0)
        else:
            res.append(1)
    elif order[0] == "front":
        if not queue:
            res.append(-1)
            continue
        res.append(queue[0])
    elif order[0] == "back":
        if not queue:
            res.append(-1)
            continue
        res.append(queue[-1])

for i in res:
    print(i)

File: java/BOJ/src/boj10866.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Deque;
import java.util.LinkedList;
import java.util.StringTokenizer;

public class boj10866 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();
        StringTokenizer stk;
        Deque<Integer> deque = new LinkedList<>();


        for (int i = 0; i < n; i++) {
            stk = new StringTokenizer(br.readLine());

            switch (stk.nextToken()) {
                case "push_back":
                    deque.addLast(Integer.parseInt(stk.nextToken()));
                    break;
                case "push_front":
                    deque.addFirst(Integer.parseInt(stk.nextToken()));
                    break;
                case "pop_front":
                    if (deque.size() == 0) {
                        sb.append("-1\n");
                    }
                    else {
                        sb.append(deque.pollFirst() + "\n");
                    }
                    break;
                case "pop_back":
                    if (deque.size() == 0) {
                        sb.append("-1\n");
                    }
                     else{
                         sb.append(deque.pollLast() + "\n");
                    }
                    break;
                case "size":
                    sb.append(deque.size() + "\n");
                    break;
                case "empty":
                    if (deque.size() != 0) {
                        sb.append("0\n");
                    } else {
                        sb.append("1\n");
                    }
                    break;
                case "front":
                    if (deque.size() == 0) {
                        sb.append("-1\n");
                    } else {
                        sb.append(deque.getFirst()+"\n");
                    }
                    break;
                case "back":
                    if (deque.size() == 0) {
                        sb.append("-1\n");
                    } else {
                        sb.append(deque.getLast() + "\n");
                    }
            }
        }
        System.out.println(sb);
    }
}
